src/azul/service/repository.py

```diff
@@ -28,10 +43,18 @@ def _get_data(self, entity_type, pagination, filters, file_url_func):
     def _get_item(self, entity_type, item_id, pagination, filters, file_url_func):
         if entity_type == 'projects':
             filters['file']['projectId'] = {"is": [item_id]}
+        elif entity_type == 'specimens':
+            filters['file']['specimenDocumentId'] = {"is": [item_id]}
         else:
             filters['file']['fileId'] = {"is": [item_id]}
+
+        try:
+            if item_id != str(uuid.UUID(item_id)):
+                raise InvalidUUIDError(item_id)
+        except ValueError:
```

Comment:
Limit the body of the `try:` block such that it only contains the invocation of `uuid.UUID()`